contracts/interchain-token-service/src/contract.rs

```diff
@@ -161,27 +161,16 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
     }
 
     /// Retrieves the address of the token associated with the specified token ID.
-    ///
-    /// # Arguments
-    /// * `env` - A reference to the environment in which the function operates.
-    /// * `token_id` - A 32-byte identifier for the token.
-    ///
-    /// # Returns
-    /// * `Address` - The address of the token associated with the given token ID.
-    fn token_address(env: &Env, token_id: BytesN<32>) -> Address {
-        Self::token_id_config(env, token_id).token_address
+    fn token_address(env: &Env, token_id: BytesN<32>) -> Result<Address, ContractError> {
+        Self::token_id_config(env, token_id).map(|config| config.token_address)
     }
 
     /// Retrieves the type of the token manager type associated with the specified token ID.
-    ///
-    /// # Arguments
-    /// * `env` - A reference to the environment in which the function operates.
-    /// * `token_id` - A 32-byte identifier for the token.
-    ///
-    /// # Returns
-    /// * `TokenManagerType` - The type of the token manager associated with the given token ID.
-    fn token_manager_type(env: &Env, token_id: BytesN<32>) -> TokenManagerType {
-        Self::token_id_config(env, token_id).token_manager_type
+    fn token_manager_type(
+        env: &Env,
+        token_id: BytesN<32>,
+    ) -> Result<TokenManagerType, ContractError> {
+        Self::token_id_config(env, token_id).map(|config| config.token_manager_type)
```

Comment:
```suggestion
    fn token_manager_type(
        env: &Env,
        token_id: BytesN<32>,
    ) -> TokenManagerType {
        Self::token_id_config(env, token_id).expect("token id config not found").token_manager_type
```
these entrypoints should still return the type as before

Reply:
updated